packages/contracts/test/contracts/discovery/Batch.ts

```diff
@@ -0,0 +1,49 @@
+describe("Batch", () => {
+  describe("constructor", () => {
+    it("works with valid arguments");
+    it("fails with an empty list of projects");
+    it("fails with an address that's not a project");
+    it("fails if one of the projects is not whitelisted");
```

Comment:
The batch needs to ask the controller for this, correct?

Reply:
can ask the project directly, I believe